# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/canvas/WebGLTypes.h
# Commit: bbfe23c61add
# Full Hash: bbfe23c61add07377daf80696fd6996c0463f5b6
# Author: Jeff Gilbert <jgilbert@mozilla.com>
# Date: 2020-06-09 09:21:34
# Regressor Bug: 1632249
# File Overlap Count: 3
# Description:
#   Bug 1632249 - Support compositing out-of-process WebGL. r=handyman,lsalzman,nical,geckoview-reviewers,agi,imanol
#   
#   * Majorly simplity CanvasRenderer
#   * Replace GLScreenBuffer with trivial GLSwapChain
#   * Use descriptor structs so that future SharedSurface changes aren't so painful
# ==============================================================================

diff -r 8fcfac450ecc -r bbfe23c61add dom/canvas/WebGLTypes.h
--- a/dom/canvas/WebGLTypes.h	Mon Jun 08 20:13:01 2020 +0000
+++ b/dom/canvas/WebGLTypes.h	Mon Jun 08 20:34:15 2020 +0000
@@ -989,6 +989,28 @@
   return {text.BeginReading(), text.Length()};
 }
 
+// -
+// ClientWebGLContext exports!
+// (So that you don't need to include the mess that is ClientWebGLContext.h)
+namespace gfx {
+class SourceSurface;
+}  // namespace gfx
+
+namespace layers {
+class KnowsCompositor;
+class SurfaceDescriptor;
+}  // namespace layers
+
+namespace webgl {
+
+void Present(ClientWebGLContext&);
+Maybe<layers::SurfaceDescriptor> GetFrontBuffer(ClientWebGLContext&,
+                                                layers::KnowsCompositor*);
+
+}  // namespace webgl
+
+// -
+
 }  // namespace mozilla
 
 #endif